bitrepository-reference-pillar/src/main/java/org/bitrepository/pillar/messagehandler/GetChecksumsRequestHandler.java

```diff
@@ -237,7 +239,8 @@ private void uploadFile(File fileToUpload, String url) throws IOException {
         // Upload the file.
         log.debug("Uploading file '{}' to {}", fileToUpload.getName(), url);
         try (InputStream in = new BufferedInputStream(new FileInputStream(fileToUpload))) {
-            context.getFileExchange().putFile(in, uploadUrl);
+            FileExchange fileExchange = FileExchangeResolver.getBasicFileExchangeFromURL(uploadUrl);
```

Comment:
Tror egentlig godt jeg kan lide statisk adgang til FileExchange, som det var før -  er der et problem med det?

Reply:
Problemet er, at FileExchange bliver oprettet ud fra Settings én enkelt gang ved opstart af pillaren, så resten af dens levetid har den kun en LocalFileExchange/Http(s)FileExchange. 

Som ønsket i de to issues, skal det gerne fungere dynamisk, så pillaren kan bruge FileExchange-funktionalitet baseret på den URL, der kommer ind fra de relevante beskeder.